feat: document `Astro.routePattern` (#9180)
withastro/docs: src/content/docs/en/reference/api-reference.mdx

The locale computed from the current URL, using the syntax specified in your `locales` configuration. If the URL does not contain a `/[locale]/` prefix, then the value will default to `i18n.defaultLocale`.

### `Astro.routePattern`

> The current route pattern being rendered. The pattern will look like a file path, without the [`srcDir`](/en/reference/configuration-reference/#srcdir) and the `pages` folder. The path will start with a leading slash, and it won't have the file extension.

Reviewer: Is a "route pattern" a known thing, like that there are docs defining it at MDN etc., or is this something we have defined? This first sentence feels like a bit of a circular definition: "routePattern is a route pattern" and that is supposed to be meaningful. :smile: It would be nice if we could link to an external source for "route pattern" if that is a standard concept. And if it's not, then we have to think of a way to describe it that is meaningful.

I'm also not clear when this would be used. Can we get an example showing actual usage? When would someone need this, and how would they use this information if they had it? Would you use this in a regular static Astro site? Is this more expected to be used by integration authors? 

Is it about build time knowledge of which file is currently being processed to create a route, or about knowing which component created the page a visitor is currently visiting? Does it require SSR?

There are lots of questions to answer here, and I think focusing on what this is used for will help!

(Also reminder we'll need a `<Since />` component, too.

The locale computed from the current URL, using the syntax specified in your `locales` configuration. If the URL does not contain a `/[locale]/` prefix, then the value will default to `i18n.defaultLocale`.

### `Astro.routePattern`

  <Since v="5.0.0" />

> The current route pattern being rendered. The value will look like a file path, without the [`srcDir`](/en/reference/configuration-reference/#srcdir) and the `pages` folder. The path will start with a leading slash, and it won't have the file extension.

Reviewer: ```suggestion

The route pattern responsible for generating the current page or route. In file-based routing, this resembles the file path in your project used to create the route. When integrations create routes for your project, `Astro.routePattern` is identical to the value for `injectRoute.pattern`.

The value will start with a leading slash and look similar to the path of a page component relative to your `srcDir/pages/` folder without a file extension.

For example, the file `src/pages/en/blog/[slug].astro` will return `/en/blog/[slug]` for `Astro.routePattern`. Every page on your site generated by that file (e.g. `/en/blog/post-1/`, /en/blog/post-2/`, etc.) shares the same value for `Astro.routePattern`.

You can use this property to understand which route is rendering your component. This allows you to target or analyze similarly-generated page URLs together. For example, you can use it to conditionally render certain information, or collect metrics about which routes are slower.
```

Hi @ematipico ! I was spending some time looking into this, and wondered whether you can tell me how accurate this description above is! Can you correct this where it's wrong? 

I was trying to add more detail to the explanation, so we don't say "routePattern is a route pattern." :smile: and also if there was a cleaner way to describe the pattern part itself without saying "excluding all these things". I also thought it might be helpful to integration authors if we mention what happens with injected routes, so I made a guess there.

So, I might not be correct here! But, can you help me by showing me what's not correct (or useful/important/helpful) with my above understanding! Thank you!

Author: Thank you, the suggestion looks good to me. I modified the suggestion to add the following phrase: 
> , `src/pages/index.astro` will return `/`

This is a particular that we should have *somewhere*, because the files that are called `index.*` don't keep the `index` inside the value of `Astro.routePattern`

Reviewer: Perfect, thank you! I just updated to give it its own sentence, but this works for me!

> 
> For example, if the current route is `src/pages/en/blog/[slug].astro`, then `Astro.routePattern` will be `/en/blog/[slug]`.
> 
> You can use this property to understand which route is rendering your component and make decisions based on it. For example, you can use it to conditionally render certain information, or collect some metrics about which routes are slower.
> 
> ```astro

Reviewer: ```suggestion
```

The locale computed from the current URL, using the syntax specified in your `locales` configuration. If the URL does not contain a `/[locale]/` prefix, then the value will default to `i18n.defaultLocale`.

### `Astro.routePattern`

> 
>   <Since v="5.0.0" />
> 

Reviewer: ```suggestion

<p>

**Type:** string <br />
  <Since v="5.0.0" />
  </p>

```
Is this correct, just trying to apply @ArmandPhilippot 's edit before he does?  :smile:

Reviewer 2: 😆 Looks good!

See also: [`Astro.locals`](#astrolocals)

### `context.routePattern`

<Since v="5.0.0" />

> The current route pattern being rendered. The value will look like a file path, without the [`srcDir`](/en/reference/configuration-reference/#srcdir) and the `pages` folder. The path will start with a leading slash, and it won't have the file extension.
> 
> For example, if the current route is `src/pages/api/insert.ts`, then `Astro.routePattern` will be `/en/blog/insert`.
> 
> You can use this property to understand which route is rendering your component and make decisions based on it. For example, you can use it to conditionally render certain information, or collect some metrics about which routes are slower.
> 

Reviewer: ```suggestion
The route pattern responsible for generating the current page or route. In file-based routing, this resembles the file path in your project used to create the route. When integrations create routes for your project, `context.routePattern` is identical to the value for `injectRoute.pattern`.
	
The value will start with a leading slash and look similar to the path of a page component relative to your `srcDir/pages/` folder without a file extension.
	
For example, the file `src/pages/en/blog/[slug].astro` will return `/en/blog/[slug]` for `context.routePattern`. Every page on your site generated by that file (e.g. `/en/blog/post-1/`, `/en/blog/post-2/`, etc.) shares the same value for `context.routePattern`. In the case of `index.*` pages, the route pattern will not include the word "index." For example, `src/pages/index.astro` will return `/`
	
You can use this property to understand which route is rendering your component. This allows you to target or analyze similarly-generated page URLs together. For example, you can use it to conditionally render certain information, or collect metrics about which routes are slower.
```

---
```

### `Astro.routePattern`

> <p>
>   **Type**: `string`<br />
>   <Since v="5.0.0" />
> </p>

Reviewer: ```suggestion
<p>

**Type**: `string`<br />
<Since v="5.0.0" />
</p>
```

---
```

### `Astro.routePattern`

<p>

**Type**: `string`<br />
<Since v="5.0.0" />
</p>

> The current route pattern being rendered. The value will look like a file path, without the [`srcDir`](/en/reference/configuration-reference/#srcdir) and the `pages` folder. The path will start with a leading slash, and it won't have the file extension.
> 
> For example, if the current route is `src/pages/en/blog/[slug].astro`, then `Astro.routePattern` will be `/en/blog/[slug]`.
> 
> You can use this property to understand which route is rendering your component and make decisions based on it. For example, you can use it to conditionally render certain information, or collect some metrics about which routes are slower.

Reviewer: ```suggestion
The route pattern responsible for generating the current page or route. In file-based routing, this resembles the file path in your project used to create the route. When integrations create routes for your project, `context.routePattern` is identical to the value for `injectRoute.pattern`.
	
The value will start with a leading slash and look similar to the path of a page component relative to your `srcDir/pages/` folder without a file extension.
	
For example, the file `src/pages/en/blog/[slug].astro` will return `/en/blog/[slug]` for `context.routePattern`. Every page on your site generated by that file (e.g. `/en/blog/post-1/`, `/en/blog/post-2/`, etc.) shares the same value for `context.routePattern`. In the case of `index.*` pages, the route pattern will not include the word "index." For example, `src/pages/index.astro` will return `/`
	
You can use this property to understand which route is rendering your component. This allows you to target or analyze similarly-generated page URLs together. For example, you can use it to conditionally render certain information, or collect metrics about which routes are slower.
```

See also: [`Astro.locals`](#astrolocals)

### `context.routePattern`

> <Since v="5.0.0" />

Reviewer: ```suggestion
<p>

**Type:** `string` <br />
  <Since v="5.0.0" />
  </p>
```
Help me out again here!

Reviewer 2: Looks good but (nit) we can remove the additional indents:
```suggestion
<p>

**Type:** `string` <br />
<Since v="5.0.0" />
</p>
```

See also [`Astro.callAction()`](#astrocallaction)

### `context.routePattern`

> <p>
>   **Type**: `string`<br />
>   <Since v="5.0.0" />
> </p>

Reviewer: ```suggestion
<p>

**Type**: `string`<br />
<Since v="5.0.0" />
</p>
```

See also [`Astro.callAction()`](#astrocallaction)

### `context.routePattern`

<p>

**Type**: `string`<br />
<Since v="5.0.0" />
</p>

> The current route pattern being rendered. The value will look like a file path, without the [`srcDir`](/en/reference/configuration-reference/#srcdir) and the `pages` folder. The path will start with a leading slash, and it won't have the file extension.
> 
> For example, if the current route is `src/pages/api/insert.ts`, then `Astro.routePattern` will be `/en/blog/insert`.
> 
> You can use this property to understand which route is rendering your component and make decisions based on it. For example, you can use it to conditionally render certain information, or collect some metrics about which routes are slower.

Reviewer: ```suggestion
The current route pattern being rendered. The value will look like a file path, without the [`srcDir`](/en/reference/configuration-reference/#srcdir) and the `pages` folder. The path will start with a leading slash, and it won't have the file extension.

For example, if the current route is `src/pages/api/insert.ts`, then `Astro.routePattern` will be `/en/blog/insert`.

You can use this property to understand which route is rendering your component and make decisions based on it. For example, you can use it to conditionally render certain information, or collect some metrics about which routes are slower.
```

> The current route pattern being rendered. The value will look like a file path, without the [`srcDir`](/en/reference/configuration-reference/#srcdir) and the `pages` folder. The path will start with a leading slash, and it won't have the file extension.
> 
> For example, if the current route is `src/pages/api/insert.ts`, then `Astro.routePattern` will be `/en/blog/insert`.
> 
> You can use this property to understand which route is rendering your component and make decisions based on it. For example, you can use it to conditionally render certain information, or collect some metrics about which routes are slower.

Reviewer: ```suggestion
The route pattern responsible for generating the current page or route. In file-based routing, this resembles the file path in your project used to create the route. When integrations create routes for your project, `Astro.routePattern` is identical to the value for `injectRoute.pattern`.

The value will start with a leading slash and look similar to the path of a page component relative to your `srcDir/pages/` folder without a file extension.

For example, the file `src/pages/en/blog/[slug].astro` will return `/en/blog/[slug]` for `Astro.routePattern`. Every page on your site generated by that file (e.g. `/en/blog/post-1/`, `/en/blog/post-2/`, etc.) shares the same value for `Astro.routePattern`. In the case of `index.*` routes, the route pattern will not include the word "index." For example, `src/pages/index.astro` will return `/`.

You can use this property to understand which route is rendering your component. This allows you to target or analyze similarly-generated page URLs together. For example, you can use it to conditionally render certain information, or collect metrics about which routes are slower.
```
